include/SFML/System/Angle.hpp

```diff
@@ -0,0 +1,473 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2022 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+#ifndef SFML_ANGLE_HPP
+#define SFML_ANGLE_HPP
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/System/Export.hpp>
+#include <cassert>
+
+
+namespace sf
+{
+////////////////////////////////////////////////////////////
+/// \brief Represents an angle value.
+///
+////////////////////////////////////////////////////////////
+class Angle
+{
+public:
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Default constructor
+    ///
+    /// Sets the angle value to zero.
+    ///
+    ////////////////////////////////////////////////////////////
+    constexpr Angle();
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Return the angle's value in degrees
+    ///
+    /// \return Angle in degrees
+    ///
+    /// \see asRadians
+    ///
+    ////////////////////////////////////////////////////////////
+    [[nodiscard]] constexpr float asDegrees() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Return the angle's value in radians
+    ///
+    /// \return Angle in radians
+    ///
+    /// \see asDegrees
+    ///
+    ////////////////////////////////////////////////////////////
+    [[nodiscard]] constexpr float asRadians() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Return the angle wrapped to the range [-180°, 180°)
+    ///
+    /// \return Angle wrapped to [-180°, 180°)
+    ///
+    /// \see wrappedTo360
+    ///
+    ////////////////////////////////////////////////////////////
+    [[nodiscard]] constexpr Angle wrappedTo180() const;
```

Comment:
What are the use cases for this functionality?

Reply:
Bromeon recommended we add this. https://github.com/SFML/SFML/pull/1969#issuecomment-1013732277

Reply:
`-180° .. 180°` is a range that represents a coordinate system where 0 is the middle, and positive/negative angles cancel each other out. It's often more intuitive, because algebra behaves close to the domain of real numbers.

Examples where it helps:
* `angle > 0` and `angle < 0` to see if an object (e.g. game entity) turns left/right
* `abs(angle)` gives you the deviation from "forward" (0)
* `-angle` is the same deviation in the opposite direction

Reply:
When cross product and maybe vector reflection will be added, I think using this `wrappedTo180` will not be so useful.

Reply:
But is there any evidence that people actually want this? What about the ones that want some other section of the circle wrapped (e.g. -90 / +90)?

Do you have any concrete examples where you implemented this yourself or wished for it to exist?

Reply:
I guess one thing I failed to really consider that those two are more or less the only ways to "wrap" a circle (symmetrically).
You either go around once, or you go "half" and "half", as such I think it starts to make more sense and there's also enough math that deals with π and -π vs 2π I can start to see different cases 🤔 

As mentioned on the other function, I'd however rethink the naming of it.